### ISSUE TITLE
spfeas test_features() error. ReadAsArray() doesn't return np.ndarray for .vrt

### ISSUE THREAD

**Issue by jwarndt** [jgrss/spfeas#29]
While running spfeas.test_features() I get the following error which traces back to mpglue's raster_tools.py. I am using spfeas 0.3.0b and mpglue 0.2.0b. Any idea what might be happening? It seems to only happen when reading a .vrt. Thanks.

![image](https://user-images.githubusercontent.com/25138755/41241826-f02c0264-6d6b-11e8-9862-1204c2131457.png)

**Comment by jgrss** [comment]
@jwarndt It looks like the spatial features were successfully created, but then the error occurred when compositing the features. 

What are the contents of `C:\Users\4ja\AppData\Local\Continuum\minicond3_64bit\envs\py35_spfeas\lib\site-packages\spfeas-0.30b0-py3.5-win-amd64.egg\spfeas\spfeas\data\_features\`?

**Comment by jwarndt** [comment]
Inside `C:\Users\4ja\AppData\Local\Continuum\minicond3_64bit\envs\py35_spfeas\lib\site-packages\spfeas-0.30b0-py3.5-win-amd64.egg\spfeas\spfeas\data\_features\`, there is:

`test_image__BD1_BK4_SC8_TRmean.vrt` 
`test_image__BD1_BK4_SC8_TRmean.yaml`  
`test_image__BD1_BK4_SC8_TRmean\test_image__BD1_BK4_SC8__ST1-002__TL000001.tif` 
`test_image__BD1_BK4_SC8_TRmean\test_image__BD1_BK4_SC8__ST1-002__TL000002.tif`

**Comment by jgrss** [comment]
Ok, and do you also have `C:\Users\4ja\AppData\Local\Continuum\minicond3_64bit\envs\py35_spfeas\lib\site-packages\spfeas-0.30b0-py3.5-win-amd64.egg\spfeas\spfeas\data\features\`?

**Comment by jwarndt** [comment]
Yes.

I also verified that the `.vrt` in both the `features` and `_features` are the same. So the tests written in `spfeas.test_features()` should pass except it gets hung up on reading one of the bands from the `.vrt` in the `features` folder.

**Comment by jgrss** [comment]
Okay, I cannot reproduce the error on Linux but I will look into the issue. In the meantime, the contextual features should run, but you won't automatically get a VRT composite after everything is finished processing.

**Comment by jgrss** [comment]
@jwarndt Can you paste the contents of the VRT file under `features`? I am wondering if there is an issue with relative paths on Windows.

**Comment by jwarndt** [comment]
The file paths appear to be correct.

![image](https://user-images.githubusercontent.com/25138755/41659426-4a223d86-7467-11e8-9f0a-0f1f646bf3eb.png)


I tested  reading bands from a .vrt with GDAL 2.2.2 (installed via `conda install gdal`). When using `ReadAsArray()` on a band of a .vrt it returns `None`.

![image](https://user-images.githubusercontent.com/25138755/41661175-19ec9fbc-746c-11e8-8a18-d613e40d8530.png)

I updated my version of GDAL to 2.2.4 and ran `spfeas.test_features()` again. This time the test completed without an error. For some reason, my previous install of GDAL 2.2.2 from conda (which I installed using `conda install gdal`) wasn't working for reading .vrts. It also wasn't working for reading BigTIFF files (Tiffs larger than 4 GB). I uninstalled the conda version and used pip to install version gdal 2.2.4. This version now works for reading BigTIFF files and .vrts. I suspect the version of gdal that is installed from the conda channel using `conda install gdal` was not built with updated supporting libraries to provide this functionality. For BigTIFFs GDAL must be built with libtiff >= 4.0 (http://www.gdal.org/frmt_gtiff.html). I'm wondering if a similar issue might be the cause for this install of GDAL not reading .vrts correctly. It might not be due to the version of GDAL, but rather how it was built. The versions of GDAL that can be installed from pip and conda-forge solves the .vrt reading issue and BigTIFF reading. So in the end, the problem was with the install of GDAL.
